### PR TITLE
Add Swagger documention for UpgradeDataWarehouse API

### PR DESCRIPTION
**Pull request by vsporeddy**
UpgradeDataWarehouse allows users to perform maintenance updates on their SQL Data Warehouse instances at their own convenience. This change adds the API specifications for this feature.

### CODE REVIEW

**Comment by marstr** [comment]
@vsporeddy these changes are to an API Version that is in `preview`. For that reason, you can opt to skip some of the normal review process. However, it would also mean that SDK owners may not generate libraries to target this service.

Is this API Version correctly sorted?

**Comment by vsporeddy** [comment]
@marstr Yes, this is for a preview API version. However I want it to be available in .NET SDK and Azure PowerShell, is that possible?

**Comment by marstr** [comment]
I see, sorry for the confusion! So, we should classify this as belonging in the `stable` folder, because it is a Swagger that will be built upon by customers. Please make that update, and we'll start the ARM review, etc.

**Comment by vsporeddy** [comment]
Oh no, I meant I was going to make it available in .NET SDK/Azure PowerShell myself. It is still only for 2017-03-01-preview.

**Comment by marstr** [comment]
> @vsporeddy says:
I meant I was going to make it available in .NET SDK/Azure PowerShell myself

Ah just for a private preview, or with the intention of shipping to customers? I think there are two things getting confused here:
1. Whether the API Version is in preview or not (should be denoted by the "-preview" suffix), and speaks to the SLA and long-term availability of that API Version.
1. Whether or not the Swagger is ready for publication to customers (outside of private preview). This is indicated by whether the API Version is in the `preview` or `stable` folder in this repository.

The two aspects are independent of one another. Clearly the API Version is under a more lenient SLA, but should it be considered ready for customers to play with?

**Comment by ravbhatnagar** [comment]
Looks good.

**Comment by marstr** [comment]
KK, well all moot at this point! @vsporeddy are you ready for me to merge this in?

**Review by marstr — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databases.json [hunk 1]
```diff
@@ -480,6 +480,52 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/upgradeDataWarehouse": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databases.json]: missing x-ms-long-running-operation?

**Review by jaredmoo — CHANGES_REQUESTED** [review]
missing x-ms-long-running-operation

**Review by jaredmoo — APPROVED** [review]

**Comment by marstr** [comment]
Just to update this thread, we're currently having some offline conversations about the `preview` folder. We should respond more adequately tomorrow with more information.

**Comment by marstr** [comment]
Sorry for the delay on this. 

After chatting about it internally with @salameer, @johanste, and @lmazuel it seems like some of my statements above had some misinformation. It sounds like we're still committed to publishing bits in the preview folder as public-previews. The only real difference between the `stable` and `preview` folders is that SDKs are required to somehow convey to consumers of SDKs generated from Swaggers in the `preivew` folder that breaking changes may come. Different SDKs will interpret that differently. [Python for instance, will release PyPI packages with a suffix to indicate preview status.](https://pypi.python.org/pypi/azure-keyvault/).

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by marstr** [comment]
@AutorestCI rebuild azure-sdk-for-go

**Comment by marstr** [comment]
Howdy @vsporeddy, with the context that I layout above, what is the current status of this PR? Is it ready to be in the `master` branch in the preview folder?

**Comment by marstr** [comment]
Closing this PR as stale. If these changes are still relevant, feel free to have it re-opened or create a new PR. One caveat, I'll be OOF starting Friday the 16th, so if it is re-opened it will need to be reassigned.

**Pull request closed without merging**